cmd/cert-checker/main_test.go

```diff
@@ -76,6 +77,52 @@ func BenchmarkCheckCert(b *testing.B) {
 	}
 }
 
+func TestCheckWildcardCert(t *testing.T) {
+	saDbMap, err := sa.NewDbMap(vars.DBConnSA, 0)
+	test.AssertNotError(t, err, "Couldn't connect to database")
+	saCleanup := test.ResetSATestDatabase(t)
+	defer func() {
+		saCleanup()
+	}()
+
+	features.Set(map[string]bool{"WildcardDomains": true})
```

Comment:
Oh yeah, it's because you are not doing a `defer features.Reset()` here, so the wildcard feature is still set for `TestCheckCert` so the `checkFunc` is working as intended and falling through to `isForbiddenDomain`.

Reply:
(Created #3377 to track fixing the package so that we don't have to keep track of this kind of thing in tests)

Reply:
I don't think that's it. I added `defer features.Reset` and it's fine. I think what happens is that errors are cumulative, not exclusive.

Reply:
That seems to have fixed the test failure I saw, you also need to catch the error here or errcheck will be mad at you, then everything should pass just fine.